src/bicome/logic/manager/GameManager.java

```diff
@@ -0,0 +1,97 @@
+package bicome.logic.manager;
+
+import java.awt.*;
+import java.awt.event.*;
+import javax.swing.*;
+import bicome.controllers.*;
+import bicome.logic.*;
+import bicome.logic.feature.*;
+import bicome.logic.world.*;
+
+/* A class that manages the game
+ * @author İsmail İlter Sezan
+ */
+public class GameManager
+{
+   
+   //properties
+   Timer turnTimer;
+   int durationOfTurns;
+   int timePassed;
+   int yearsPassed;
+   int numOfTurns;
+   World world;
+   GameController controller;
+   FeatureList features;
+   
+   //constructor
+   public GameManager( World world , GameController controller )
```

Comment:
Why you have controller in the constructor?

Reply:
Because controller gets the initial location

Reply:
Controller should have the gameManager not the other way.

Reply:
Why then we have those codes which gets the click is in controller then?

Reply:
Just to remind: we decided to it will take the controller in order to call the refreshing function of the page. I was forgot that one.